src/server.c

```diff
@@ -5158,682 +5158,751 @@ dict *genInfoSectionDict(robj **argv, int argc, char **defaults, int *out_all, i
+    atomicGet(server.lruclock,lruclock);
+    info = sdscatfmt(info,
+        "redis_version:%s\r\n"
+        "redis_git_sha1:%s\r\n"
+        "redis_git_dirty:%i\r\n"
+        "redis_build_id:%s\r\n"
+        "redis_mode:%s\r\n"
+        "os:%s %s %s\r\n"
+        "arch_bits:%i\r\n"
+        "monotonic_clock:%s\r\n"
+        "multiplexing_api:%s\r\n"
+        "atomicvar_api:%s\r\n"
+        "gcc_version:%i.%i.%i\r\n"
+        "process_id:%I\r\n"
+        "process_supervised:%s\r\n"
+        "run_id:%s\r\n"
+        "tcp_port:%i\r\n"
+        "server_time_usec:%I\r\n"
+        "uptime_in_seconds:%I\r\n"
+        "uptime_in_days:%I\r\n"
+        "hz:%i\r\n"
+        "configured_hz:%i\r\n"
+        "lru_clock:%u\r\n"
+        "executable:%s\r\n"
+        "config_file:%s\r\n"
+        "io_threads_active:%i\r\n",
+        REDIS_VERSION,
+        redisGitSHA1(),
+        strtol(redisGitDirty(),NULL,10) > 0,
+        redisBuildIdString(),
+        mode,
+        name.sysname, name.release, name.machine,
+        server.arch_bits,
+        monotonicInfoString(),
+        aeGetApiName(),
+        REDIS_ATOMIC_API,
 #ifdef __GNUC__
-            __GNUC__,__GNUC_MINOR__,__GNUC_PATCHLEVEL__,
+        __GNUC__,__GNUC_MINOR__,__GNUC_PATCHLEVEL__,
 #else
-            0,0,0,
+        0,0,0,
 #endif
-            (int64_t) getpid(),
-            supervised,
-            server.runid,
-            server.port ? server.port : server.tls_port,
-            (int64_t)server.ustime,
-            (int64_t)uptime,
-            (int64_t)(uptime/(3600*24)),
-            server.hz,
-            server.config_hz,
-            lruclock,
-            server.executable ? server.executable : "",
-            server.configfile ? server.configfile : "",
-            server.io_threads_active);
-
-        /* Conditional properties */
-        if (isShutdownInitiated()) {
-            info = sdscatfmt(info,
-                "shutdown_in_milliseconds:%I\r\n",
-                (int64_t)(server.shutdown_mstime - server.mstime));
-        }
+        (int64_t) getpid(),
+        supervised,
+        server.runid,
+        server.port ? server.port : server.tls_port,
+        (int64_t)server.ustime,
+        (int64_t)uptime,
+        (int64_t)(uptime/(3600*24)),
+        server.hz,
+        server.config_hz,
+        lruclock,
+        server.executable ? server.executable : "",
+        server.configfile ? server.configfile : "",
+        server.io_threads_active);
+
+    /* Conditional properties */
+    if (isShutdownInitiated()) {
+        info = sdscatfmt(info,
+            "shutdown_in_milliseconds:%I\r\n",
+            (int64_t)(server.shutdown_mstime - server.mstime));
     }
+    return info;
+}
 
-    /* Clients */
-    if (all_sections || (dictFind(section_dict,"clients") != NULL)) {
-        size_t maxin, maxout;
-        getExpansiveClientsInfo(&maxin,&maxout);
-        if (sections++) info = sdscat(info,"\r\n");
+sds genRedisInfoStringClients(sds info) {
```

Comment:
I'm not a very big fan of the refactoring as it obstructs a lot of the git history.

Reply:
i also don't see a reason for this refactory just to add a few lines.

also note that you don't append to the same `info` string, instead you crate a new temp file and then append that one to the other.
this could have performance implications (more mallocs and reallocs)

Reply:
ohh, scrap that, i was looking at it in reverse, and also didn't notice the changes in server.c.

the changes in server.c gives me the shivers.
maybe there's a way to do that without actually changing the lines (keep the indentation and ordering).
but i'm not sure there's any real benefit of breaking up this huge function either.

Reply:
Hi Oran, the only reason I want to refactor it is the function length is too long, it includes more than 800 lines in one single function.  After refactoring it, the single function is shorter and part of function can be reused.

Reply:
long functions (of 800 lines) is generally bad, if they have a lot of logic that is better be split, in this case it's a "script" (i.e. "do this, then this, then this..."), so i'm not sure it's that bad.
i also don't mind breaking it up, but i'd like to find a way to do that without affecting all the lines (i.e. if we can do that in a way that the diff will not change all the lines). if we somehow keep the same indentation, and order, maybe there's a way.